frontend/src/app/projects/create-project/create-project.component.ts

```diff
@@ -3,49 +3,149 @@
+  private projectsSlugs = new BehaviorSubject<string[]>([]);
+  private slugsSubscription?: Subscription;
 
-  get name(): FormControl {
-    return this.createProjectForm.get('name') as FormControl;
+  constructor(
+    public projectService: ProjectService,
+    private router: Router,
+    private toastService: ToastService,
+    private navBarService: NavBarService
+  ) {
+    this.navBarService.title = 'Create Project';
   }
 
-  constructor(public projectService: ProjectService, private router: Router) {}
-
   ngOnInit(): void {
-    this.projectService.project = undefined;
+    this.form.controls.name.addValidators(this.slugValidator.bind(this));
+
+    let projects = this.projectService._projects;
+
+    this.projectService.list().subscribe({
+      next: (value) => {
+        projects.next(value);
+      },
+      error: (_) => {
+        projects.next(undefined);
+      },
+    });
+
+    this.slugsSubscription = projects
+      .pipe(
+        filter(Boolean),
+        map((projects) => projects.map((p) => p.slug))
+      )
+      .subscribe(this.projectsSlugs);
+  }
+
+  ngOnDestroy(): void {
+    this.slugsSubscription?.unsubscribe();
+    if (!this.projectDetails) {
+      this.projectService._project.next(undefined);
+    }
   }
 
   createProject(stepper: MatStepper): void {
-    if (this.createProjectForm.valid) {
-      this.projectService
-        .createProject(this.name.value)
+    let projectSubject = this.projectService._project;
+    if (this.form.valid) {
+      const projectConnectable = connectable<Project>(
+        this.projectService.createProject({
+          name: this.form.value.name!,
+          description: this.form.value.name!,
+        }),
+        {
+          connector: () => new Subject(),
+          resetOnDisconnect: false,
+        }
+      );
+
+      projectConnectable
+        .pipe(
+          tap({
+            next: (value) => {
+              projectSubject.next(value);
+            },
+            error: (_) => {
+              projectSubject.next(undefined);
+            },
+          })
+        )
         .subscribe((project) => {
-          this.project_slug = project.slug;
+          this.toastService.showSuccess(
+            `The project “${project!.name}” was successfuly created.`,
+            'Project created'
+          );
+          stepper.steps.get(0)!.completed = true;
           stepper.next();
+          stepper.steps.get(0)!.editable = false;
         });
+
+      projectConnectable
+        .pipe(switchMap(() => this.projectService.list()))
+        .subscribe({
+          next: (value) => {
+            this.projectService._projects.next(value);
+          },
+          error: (_) => {
+            this.projectService._projects.next(undefined);
+          },
+        });
+
+      projectConnectable.connect();
+    }
+  }
+
+  finish(): void {
+    this.projectDetails = true;
+    this.router.navigate(['/project', this.projectService.project!.slug]);
+  }
```

Comment:
This should be used actually, instead of a simple redirect to the details. The goal is to prevent ProjectService.project to be set to undefined when `CreateProjectComponent` is destroyed, if redirected to the project details. I should replace the _Finish_ link with a button that calls `CreateProjectComponent.finish`.

Reply:
Fixed.